### PR TITLE
PBKDF2: Out of range input cost

### LINKED ISSUE #7653: Make out-of-range INPUT_COST test case for PBKDF2 use PSA_VENDOR_PBKDF2_MAX_ITERATIONS
https://github.com/Mbed-TLS/mbedtls/pull/7547 added a test case for PBKDF2 with an input cost out of range, but the test case as written can't depend on the macro that defines the bound, it has a (wrong) value hard-coded. Fixing this in #7547 is impractical because it requires a more recent development.

The goal of this issue is to implement https://github.com/Mbed-TLS/mbedtls/pull/7547#discussion_r1204581917.

### PR DESCRIPTION
**Pull request by silabs-Kusumit**
## Description

Add test function and case for testing correct behavior when input cost is greater than `PSA_VENDOR_PBKDF2_MAX_ITERATIONS`.
Also add test case when input cost is zero

Fixes #7653 

## PR checklist

- [x] **changelog**  not required, testing only
- [x] **backport** not required - not in 2.28
- [x] **tests** provided

### CODE REVIEW

File: tests/suites/test_suite_psa_crypto.data [hunk 1]
```diff
@@ -5540,6 +5535,10 @@ PSA key derivation: PBKDF2-HMAC-SHA256, reject seed step
 depends_on:PSA_WANT_ALG_PBKDF2_HMAC:PSA_WANT_ALG_SHA_256
 derive_input:PSA_ALG_PBKDF2_HMAC(PSA_ALG_SHA_256):PSA_KEY_DERIVATION_INPUT_SEED:PSA_KEY_TYPE_NONE:"":PSA_ERROR_INVALID_ARGUMENT:PSA_KEY_DERIVATION_INPUT_SALT:PSA_KEY_TYPE_NONE:"73616c74":PSA_ERROR_BAD_STATE:PSA_KEY_DERIVATION_INPUT_PASSWORD:PSA_KEY_TYPE_NONE:"706173737764":PSA_ERROR_BAD_STATE:PSA_KEY_TYPE_NONE:PSA_ERROR_BAD_STATE
 
+PSA key derivation: PBKDF2-HMAC-SHA256, reject cost greater than PSA_VENDOR_PBKDF2_MAX_ITERATIONS
+depends_on:PSA_WANT_ALG_PBKDF2_HMAC:PSA_WANT_ALG_SHA_256
+derive_input_invalid_cost:PSA_ALG_PBKDF2_HMAC(PSA_ALG_SHA_256):PSA_VENDOR_PBKDF2_MAX_ITERATIONS+1ULL
+
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_psa_crypto.data:5541]: Please add a test case with cost=0, I just realized we're missing that.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Looks good to me for the original scope, but I just realized we're missing a test case where the cost is too low (PKCS#5 says the cost must be >0) so please add that.

**Comment by gilles-peskine-arm** [comment]
There's a known issue with the GitHub reporting from OpenCI at the moment. https://ci.trustedfirmware.org/blue/organizations/jenkins/mbed-tls-pr-head/detail/PR-7702-head/2/pipeline/ passed on cbfe333c2b78b422ccd85269e02d5083ebd81eee.

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Review by daverodgman — APPROVED** [review]
